### PR TITLE
Maps birth sex values as expected by enrollment service upstream

### PR DESCRIPTION
**Pull request by patrickvinograd**
## Description of change
Updates CSV generation for to map values to "M","F",or empty string as expected by enrollment service. 

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/22643

### CODE REVIEW

**Review by batemapf — APPROVED** [review]
LGTM from a product standpoint. Really know nil about `nil`, but going to assume it is rendered as nothing in the caret delimited file :-)

**Comment by patrickvinograd** [comment]
> LGTM from a product standpoint. Really know nil about `nil`, but going to assume it is rendered as nothing in the caret delimited file :-)

Yep, the spec verifies that nil gets mapped as empty e.g. `^123456789^^` (using the known ssn value as an anchor for the following empty field).

File: modules/covid_vaccine/app/services/covid_vaccine/v0/expanded_registration_csv_generator.rb [hunk 1]
```diff
@@ -17,13 +17,19 @@ class ExpandedRegistrationCsvGenerator
         r.raw_form_data['preferred_facility']&.delete_prefix('vha_')
       end
 
+      birth_sex_proc = proc do |r|
+        next r.raw_form_data['birth_sex'][0] if %w[Male Female].include? r.raw_form_data['birth_sex']
```
Comment (saneshark) [inline modules/covid_vaccine/app/services/covid_vaccine/v0/expanded_registration_csv_generator.rb:21]: how about 

How about:

```suggestion
        { 'Male' => 'M', 'Female' => 'F' }.fetch(r.raw_form_data['birth_sex'][0].upcase, nil)
```

or 

```suggestion
        %w[M F]).find(r.raw_form_data['birth_sex'][0].upcase)
```

Comment (saneshark) [reply modules/covid_vaccine/app/services/covid_vaccine/v0/expanded_registration_csv_generator.rb:21]: might not even need the `upcase` if it's always one of those three values

**Review by saneshark — APPROVED** [review]
Nitpick comments to simplify / make more readable, but logic and specs are sound as is.

**Pull request merged**